libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -438,23 +438,37 @@ const AP_Param::GroupInfo AP_InertialSensor::var_info[] = {
 
 AP_InertialSensor *AP_InertialSensor::_s_instance = nullptr;
 
-AP_InertialSensor::AP_InertialSensor() :
-    _gyro_count(0),
-    _accel_count(0),
-    _backend_count(0),
-    _accel(),
-    _gyro(),
-    _board_orientation(ROTATION_NONE),
-    _primary_gyro(0),
-    _primary_accel(0),
-    _log_raw_bit(-1),
-    _hil_mode(false),
-    _calibrating(false),
-    _backends_detected(false),
-    _dataflash(nullptr),
-    _accel_cal_requires_reboot(false),
-    _startup_error_counts_set(false),
-    _startup_ms(0)
+
+AP_InertialSensor::AP_InertialSensor()
+#if defined(HAL_BOARD_SHOULD_ZERO_MEMBERS) && HAL_BOARD_SHOULD_ZERO_MEMBERS
```

Comment:
this is only used here, should it be also used in all other constructor?

Reply:
@khancyr this is part of "The last 4 commits" that don't really belong to this PR (https://github.com/ArduPilot/ardupilot/pull/6779#issuecomment-326111104).

Reply:
i.e. yes, it should be done in all other constructors and I will do them on top in a separate PR. 95 commits seemed a lot already :)

Reply:
ok! I didn't saw it was part of the last 4. otherwise it is nicely done, I don't have Fransisco eyes but I didn't spot mistake